src/tests/agent.ts

```diff
@@ -81,6 +81,31 @@ export default function agentTests(testContext: TestContext) {
                 const getInvalidDid = await ad4mClient.agent.byDID("na");
                 expect(getInvalidDid).toBe(null);
             })
+            it('can create entanglementProofPreFlight', async () => {
+                const ad4mClient = testContext.ad4mClient!;
+
+                //Check can generate a preflight key
+                const preFlight = await ad4mClient.agent.entanglementProofPreFlight("ethAddr");
+                expect(preFlight.deviceKey).toBe("ethAddr");
+                expect(preFlight.didSignedByDeviceKey).toBeNull();
+
+                //Check can save a entanglement proof
+                preFlight.didSignedByDeviceKey = "ethSignedDID";
+                const addProof = await ad4mClient.agent.addEntanglementProofs([preFlight as EntanglementProofInput]);
+                expect(addProof[0]).toStrictEqual(preFlight);
+
+                //Check can get entanglment proofs
+                const getProofs = await ad4mClient.agent.getEntanglementProofs();
+                expect(getProofs[0]).toStrictEqual(preFlight);
+
+                //Check can delete entanglement proofs
+                const deleteProofs = await ad4mClient.agent.deleteEntanglementProofs([preFlight as EntanglementProofInput]);
+                expect(deleteProofs.length).toBe(0);
+
+                //Check entanglement proof is deleted on get
+                const getProofsPostDelete = await ad4mClient.agent.getEntanglementProofs();
+                expect(getProofsPostDelete.length).toBe(0);
+            })
```

Comment:
Hm, to really test what is important, we should check the created signature (from `entanglementProofPreFlight()`) to be valid with regards to the agent's DID...

Reply:
I would suggest to move the new signing code from `AgentService.ts` to `Signature.ts` and have a `verify()` function there as well that does not add the Expression cludder/wrapper so that we could call that here to verify the signature..